src/main/kotlin/me/hubertus248/deployer/applications/model/entity/ApplicationName.kt

```diff
@@ -0,0 +1,20 @@
+package me.hubertus248.deployer.applications.model.entity
+
+import me.hubertus248.deployer.exception.BadRequestException
+import javax.persistence.Column
+import javax.persistence.Embeddable
+
+@Embeddable
+data class ApplicationName(
+    @Column(name = "name", length = 128, nullable = false, unique = true, updatable = false)
+    val value: String
+) {
+    init {
+        try {
+            require(value.isNotBlank())
+            require(value.all { it.isLetterOrDigit() || "_ -".contains(it) })
+        } catch (e: IllegalArgumentException) {
+            throw BadRequestException()
```

Comment:
Consider some messages to tell where the verification failed

Reply:
It's going to be a bit complicated to do properly, let's create an issue for now #9.